CRM/Export/BAO/ExportProcessor.php

```diff
@@ -1109,7 +1109,20 @@ public function getTransformedFieldValue($field, $iterationDAO, $fieldValue, $me
     ) {
       //check for custom data
       if ($cfID = CRM_Core_BAO_CustomField::getKeyID($field)) {
-        return CRM_Core_BAO_CustomField::displayValue($fieldValue, $cfID);
+        $html_type = CRM_Core_DAO::getFieldValue('CRM_Core_DAO_CustomField', $cfID, 'html_type');
+        $displayValue = CRM_Core_BAO_CustomField::displayValue($fieldValue, $cfID);
+        //need to calculate the link to the file for file custom data
+        if ($html_type === 'File') {
+          preg_match('/<a[^>]+href=([\'"])(?<href>.+?)\1[^>]*>/i', $displayValue, $matches);
```

Comment:
Nice regex! But I'm thinking to call `CRM_Core_BAO_File::getEntityFile('*', $fieldValue)` instead which has a `url` member you can use instead of parsing it out, and then only need to call CustomField::displayValue if it's not html_type File.

Reply:
@demeritcowboy 
It does calculate/return bunch of stuff that we don't want including tags for each of the rows
 ```
   [32] => Array
        (
            [fileID] => 32
            [entityID] => 32
            [mime_type] => image/jpeg
            [fileName] => download_3783e501e70d7a89b8d982f999b02b0c.jpeg
            [description] => 
            [cleanName] => download.jpeg
            [fullPath] => /home/latest/public_html/sites/default/files/civicrm/custom//download_3783e501e70d7a89b8d982f999b02b0c.jpeg
            [url] => /civicrm/file?reset=1&amp;id=32&amp;eid=531&amp;fcs=d4408436a59726fa0b19a7e1c613156d75fecf81181018078659a98996df690c_1591273757_168
            [href] => <a href="/civicrm/file?reset=1&amp;id=32&amp;eid=531&amp;fcs=d4408436a59726fa0b19a7e1c613156d75fecf81181018078659a98996df690c_1591273757_168">download.jpeg</a>
            [tag] => 
            [icon] => fa-file-image-o
        )

```

It might make sense to just put this part in a method where one just calculates the link or have the same function `getEntityFile` not calculate other bits.